qiime/sdk/artifact.py

```diff
@@ -185,8 +202,16 @@ def _parse_uuid(self, string):
         return uuid.UUID(hex=string)
 
     # TODO implement provenance parsing
-    def _parse_provenance(self, string):
-        return string
+    def _parse_provenance(self, provenance):
+        if isinstance(provenance, str):
```

Comment:
This is handling the case where there's no provenance in the artifact. Currently if there is no provenance, the serialized artifact provenance is the string `'No provenance available because this artifact was generated independently of QIIME.'`. We're still working out provenance and importing data into an artifact so this is just a placeholder for now (#26, #31).